onnxruntime/core/graph/graph_utils.cc

```diff
@@ -42,6 +42,17 @@ bool IsSupportedOptypeVersionAndDomain(const Node& node,
   return true;
 }
 
+bool IsSupportedProvider(const Node& node,
+                         const std::vector<std::string>& compatible_providers) {
```

Comment:
>const std::vector<std::string>& compatible_providers [](start = 25, length = 52)

Would unordered_set be a better data structure given we want to check if the single value from a node is in this collection?